colorizer_data/tests/test_writer.py

```diff
@@ -270,41 +271,161 @@ def test_writer_overwrites_duplicate_backdrop_keys(tmp_path):
+                assert feature_data["data"] == [0, 1, 2, 3, 4]
 
-    feature_info = FeatureInfo(key="feature", label="Feature")
-    writer.write_feature(np.array([0, 1, 2, 3, 4]), feature_info, min=-5, max=3)
-    writer.write_manifest()
+    def test_write_feature_writes_parquet_data(self, tmp_path):
+        writer = ColorizerDatasetWriter(tmp_path, DEFAULT_DATASET_NAME)
+        setup_dummy_writer_data(writer)
 
-    with open(tmp_path / DEFAULT_DATASET_NAME / "manifest.json", "r") as f:
-        manifest: DatasetManifest = json.load(f)
-        feature_file = manifest["features"][0]["data"]
-        with open(tmp_path / DEFAULT_DATASET_NAME / feature_file, "r") as f2:
-            feature_data = json.load(f2)
-            assert feature_data["min"] == -5
-            assert feature_data["max"] == 3
-            assert feature_data["data"] == [0, 1, 2, 3, 4]
+        feature_info = FeatureInfo(key="feature", label="Feature")
+        data = np.array([-5, 0, 2, 3, 4])
+        writer.write_feature(data, feature_info)
+        writer.write_manifest()
+
+        with open(tmp_path / DEFAULT_DATASET_NAME / "manifest.json", "r") as f:
+            manifest: DatasetManifest = json.load(f)
+            feature_info = manifest["features"][0]
+            # Min and max should be saved to the manifest
+            assert feature_info["min"] == -5
+            assert feature_info["max"] == 4
+
+            # Data should be saved to a parquet file
+            feature_file = feature_info["data"]
+            assert feature_file.endswith(".parquet")
+            assert os.path.exists(tmp_path / DEFAULT_DATASET_NAME / feature_file)
+
+            # Check parquet data has expected contents
+            table = pq.read_table(tmp_path / DEFAULT_DATASET_NAME / feature_file)
+            assert table.to_pandas()["data"].tolist() == data.tolist()
+
+    def write_features_can_write_nan_to_parquet(self, tmp_path):
+        writer = ColorizerDatasetWriter(tmp_path, DEFAULT_DATASET_NAME)
+        setup_dummy_writer_data(writer)
+
+        feature_info = FeatureInfo(key="feature", label="Feature")
+        data = np.array([np.nan, 1, 2, np.nan, 4])
+        writer.write_feature(data, feature_info)
+        with open(tmp_path / DEFAULT_DATASET_NAME / "manifest.json", "r") as f:
+            manifest: DatasetManifest = json.load(f)
+            feature_info = manifest["features"][0]
+            assert feature_info["min"] == 1
+            assert feature_info["max"] == 4
+
+            # Data should be saved to a parquet file
+            feature_file = feature_info["data"]
+            # Check parquet data has expected contents
+            table = pq.read_table(tmp_path / DEFAULT_DATASET_NAME / feature_file)
+            file_data = table.to_pandas()["data"].tolist()
+            assert np.isnan(file_data[0])
+            assert file_data[1] == 1
+            assert file_data[2] == 2
+            assert np.isnan(file_data[3])
+            assert file_data[4] == 4
```

Comment:
Can't do a direct comparison with the original `data` array because `nan != nan`